src/__tests__/index.tests.ts

```diff
@@ -1,78 +1,421 @@
 import pGraph from "../index";
-import { DepGraphArray } from "../types";
+import { PGraphNodeMap, PGraphNode, DependencyList } from "../types";
 
-describe("Public API", () => {
-  let calls = [];
+interface MockFunctionDefinition {
+  /** A friendly name for the function */
+  name: string;
 
-  // Example graph from: https://www.npmjs.com/package/toposort
-  const putOnShirt = () =>
-    Promise.resolve("put on your shirt").then((v) => {
-      calls.push(v);
-    });
-  const putOnShorts = () =>
-    Promise.resolve("put on your shorts").then((v) => {
-      calls.push(v);
-    });
-  const putOnJacket = () =>
-    Promise.resolve("put on your jacket").then((v) => {
-      calls.push(v);
-    });
-  const putOnShoes = () =>
-    Promise.resolve("put on your shoes").then((v) => {
-      calls.push(v);
+  /** How many ticks this function should take to simulate the duration of the function execution */
+  duration: number;
```

Comment:
We're not tackling duration in this PR, let's remove it for now

Reply:
This is just for tests, it's to control when the promises resolve relative to one another